app/Http/Controllers/API/v1/StatusTagController.php

```diff
@@ -0,0 +1,310 @@
+     *
+     * @return JsonResponse
+     */
+    public function index(int $statusId): JsonResponse {
+        try {
+            $status = Status::findOrFail($statusId);
+            return $this->sendResponse(
+                data: StatusTagResource::collection(StatusTagBackend::getVisibleTagsForUser($status, auth()->user())),
+            );
+        } catch (ModelNotFoundException) {
+            return $this->sendError(
+                error: 'No status found for this id',
+            );
+        }
+    }
+
+    /**
+     * @OA\Put(
+     *      path="/status/{statusId}/tags/{tagKey}",
+     *      operationId="updateSingleStatusTag",
+     *      tags={"Status"},
+     *      summary="Update a StatusTag",
+     *      description="Updates a single StatusTag Object, if user is authorized to",
+     *      @OA\Parameter (
+     *          name="statusId",
+     *          in="path",
+     *          description="Status-ID",
+     *          example=1337,
+     *          @OA\Schema(type="integer")
+     *      ),
+     *      @OA\Parameter (
+     *          name="tagKey",
+     *          in="path",
+     *          description="Key of StatusTag",
+     *          example="seat",
+     *          @OA\Schema(type="string")
+     *      ),
+     *      @OA\RequestBody(
+     *          @OA\MediaType(
+     *              mediaType="application/json",
+     *              @OA\Schema(
+     *                  ref="#/components/schemas/StatusTag"
+     *              ),
+     *          ),
+     *      ),
+     *      @OA\Response(
+     *          response=200,
+     *          description="successful operation",
+     *          @OA\JsonContent(
+     *              @OA\Property (
+     *                  property="data",
+     *                  type="object",
+     *                  ref="#/components/schemas/StatusTag"
+     *              )
+     *          )
+     *      ),
+     *      @OA\Response(response=400, description="Bad request"),
+     *      @OA\Response(response=401, description="Unauthorized"),
+     *      @OA\Response(response=404, description="No status found for this id"),
+     *      @OA\Response(response=403, description="User not authorized to manipulate this status"),
+     *      security={
+     *          {"passport": {"write-statuses"}}, {"token": {}}
+     *      }
+     *     )
+     *
+     * @param Request $request
+     * @param int     $statusId
+     * @param string  $tagKey
+     *
+     * @return JsonResponse
+     * @throws ValidationException
+     */
+    public function update(Request $request, int $statusId, string $tagKey): JsonResponse {
+        $validated = $request->validate([
+                                            'key'        => ['nullable', 'string', 'max:255'],
+                                            'value'      => ['required', 'string', 'max:255'],
+                                            'visibility' => ['nullable', Rule::in(StatusVisibility::keys())],
```

Comment:
Please use the correct ENUM implementation of the validator.
See: https://laravel.com/docs/9.x/validation#rule-enum
```suggestion
                                            'visibility' => ['nullable', new Enum(StatusVisibility)],
```

Reply:
I wanted to check against the keys. That's why I am using this.
But I will change this so users need to give the enum value with the request.